File: api/HomeBudget.API/HomeBudget.Application/DTOs/TransacaoDtos/TransacaoDto.cs
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.TransacaoDtos
{
    public class TransacaoDto
    {
        public Guid Id { get; set; }
        public string Categoria { get; set; }
        public string Pessoa { get; set; }

        public string Descricao { get; set; }

        public TipoTransacao Tipo { get; set; }

        public double Receitas { get; set; }
        public double Despesas { get; set; }

    }
}

File: web/src/interfaces/TransacaoDtos/TransacaoDto.ts
import { TipoTransacao } from "../../enums/TipoTransacaoEnum";

export class TransacaoDto {
  id: string;
  categoria: string;
  pessoa: string;
  descricao: string;
  tipo: TipoTransacao;
  receitas: number;
  despesas: number;

  constructor(
    id: string,
    categoria: string,
    pessoa: string,
    descricao: string,
    tipo: TipoTransacao,
    receitas: number,
    despesas: number
  ) {
    this.id = id;
    this.categoria = categoria;
    this.pessoa = pessoa;
    this.descricao = descricao;
    this.tipo = tipo;
    this.receitas = receitas;
    this.despesas = despesas;
  }
}